packages/magento-product/components/AddProductsToCart/useAddProductsToCartAction.ts

```diff
@@ -33,6 +37,31 @@ export function useAddProductsToCartAction(
     loading,
   } = props
 
+  const [showSuccess, setShowSuccess] = useState<boolean>(false)
+
+  const userErrors = toUserErrors(data)
+
+  const submitSuccesful =
+    !formState.isSubmitting &&
+    formState.isSubmitSuccessful &&
+    !error?.message &&
+    !userErrors.length &&
+    !redirect
```

Comment:
Do we want to only enable this behavior when there is `!redirect` or do we always want to enable this behavior?